packages/grid/x-data-grid/src/colDef/gridMultipleSelectColDef.tsx

```diff
@@ -0,0 +1,44 @@
+import { isArrayOfObjects } from './gridSelectColDefUtils';
+import { GRID_SINGLE_SELECT_COL_DEF } from './gridSingleSelectColDef';
+import { GridMultipleSelectColDef, ValueOptions } from '../models/colDef/gridColDef';
+import { renderEditMultipleSelectCell } from '../components/cell/GridEditMultipleSelectCell';
+import { getGridMultipleSelectOperators } from './gridMultipleSelectOperators';
+import { isMultipleSelectColDef } from '../components/panel/filterPanel/filterPanelUtils';
+
+export const GRID_MULTIPLE_SELECT_COL_DEF: Omit<GridMultipleSelectColDef, 'field'> = {
+  ...GRID_SINGLE_SELECT_COL_DEF,
+  type: 'multipleSelect',
+  valueFormatter(params) {
+    const { id, field, value, api } = params;
+    const colDef = params.api.getColumn(field);
+
+    if (!isMultipleSelectColDef(colDef)) {
+      return [];
+    }
```

Comment:
Is this check necessary if the formatter is for this column?

Reply:
This formatter returns `[]` in a few places, isn't it supposed to return a string?

Reply:
Good question - it's a continuation of Jim's work. I'd agree that streamlining the return type here makes sense.

Reply:
https://mui.com/material-ui/api/autocomplete/#autocomplete-prop-multiple - we explicitly set multiple on the autocomplete component, but that I would assume would mean that the succeeding string branches should be arrays as well.

Reply:
But IIUC, `valueFormatter` is only for formatting to string. One of its use-cases is filtering (full-text search). I'm not sure how else it's used. Maybe someone else can add more context.

Reply:
https://next.mui.com/x/react-data-grid/column-definition/#value-formatter says:

> Note, that the value returned by valueFormatter is only used for rendering purposes. Filtering and sorting are based on the raw value (row[field]) or the value returned by [valueGetter](https://next.mui.com/x/react-data-grid/column-definition/#value-getter).

But this is wrong, the quickfilter code uses the formatted values for its filtering.